### PR TITLE
build(devtools): node 16

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automated conversion to node16 module resoultion.

[AB#7225](https://dev.azure.com/fluidframework/internal/_workitems/edit/7225)

### CODE REVIEW

File: packages/tools/devtools/devtools/package.json [hunk 1]
```diff
@@ -37,7 +50,7 @@
 		"prettier:fix": "prettier --write . --cache --ignore-path ../../../../.prettierignore",
 		"rebuild": "npm run clean && npm run build",
 		"test": "echo TODO: add tests",
-		"tsc": "tsc",
+		"tsc": "tsc-multi --config ../../../../common/build/build-common/tsc-multi.node16.cjs.json && copyfiles -f ../../../../common/build/build-common/src/cjs/package.json ./dist",
```
Comment (jason-ha) [inline packages/tools/devtools/devtools/package.json:53]: move to fluid-tsc...

File: packages/tools/devtools/devtools/package.json [hunk 2]
```diff
@@ -37,7 +50,7 @@
 		"prettier:fix": "prettier --write . --cache --ignore-path ../../../../.prettierignore",
 		"rebuild": "npm run clean && npm run build",
 		"test": "echo TODO: add tests",
-		"tsc": "tsc",
+		"tsc1": "fluid-tsc commonjs --project ./tsconfig.cjs.json && copyfiles -f ../../../common/build/build-common/src/cjs/package.json ./dist",
```
Comment (jason-ha) [inline packages/tools/devtools/devtools/package.json:53]: typo? `tsc1` -> `tsc`

File: packages/tools/devtools/devtools/package.json [hunk 3]
```diff
@@ -91,6 +105,7 @@
 		"moment": "^2.21.0",
 		"prettier": "~3.0.3",
 		"rimraf": "^4.4.0",
+		"tsc-multi": "^1.1.0",
```
Comment (jason-ha) [inline packages/tools/devtools/devtools/package.json:108]: remove

File: pnpm-lock.yaml [hunk 4]
```diff
@@ -18167,7 +18171,7 @@ packages:
     dependencies:
       '@jest/fake-timers': 29.7.0
       '@jest/types': 29.6.3
-      '@types/node': 18.19.1
+      '@types/node': 20.10.5
```
Comment (jason-ha) [inline pnpm-lock.yaml:18174]: should also e rolled back

File: packages/tools/devtools/devtools/tsconfig.cjs.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+	// This config must be used in a "type": "commonjs" environment. (Use tsc-mult.)
```
Comment (jason-ha) [inline packages/tools/devtools/devtools/tsconfig.cjs.json:2]: `fluid-tsc commonjs`

**Review by jason-ha — COMMENTED** [review]
Is this branch far behind main by any chance?  .... I guess not maybe I wasn't looking at all diffs at one point.

**Review by tylerbutler — APPROVED** [review]

**Pull request merged**